### PR TITLE
ci: Update shellcheck and build image hashes

### LINKED ISSUE #13228: Update shellcheck in CI
## description

Currently we use the `shellcheck` (v 0.4.6-1) provided by Ubuntu `bionic` in  CI.

This is quite out of date - shellcheck has improved with each version, and its different to the versions found at least on ~recent linux desktops (not sure about mac/windows ?)

As #13106 lands all of the code will be brought under shellcheck coverage, and already some changes have landed which would have failed in a newer version, so it would be good in terms of maintaining the linting to upgrade this package.

I dont see any ~official PPA with a backport - so i guess one of

- using upstream directly
- using a docker image

...will be the easiest way to upgrade

## refs

- https://www.ubuntuupdates.org/ppas
- https://github.com/koalaman/shellcheck
- https://hub.docker.com/r/koalaman/shellcheck

### PR DESCRIPTION
**Pull request by phlax**
Commit Message: ci: Update shellcheck and build image hashes
Additional Description:

some shellcheck cleanups that are required once shellcheck is updated

this pr needs to /wait until envoyproxy/envoy-build-tools#103 lands and to have the new build image hashes added

Risk Level: low
Testing:
Docs Changes:
Release Notes:
[Optional Runtime guard:]
[Optional Fixes #Issue] Fixes #13228 
[Optional Deprecated:]

### CODE REVIEW

File: tools/code_format/check_shellcheck_format.sh [hunk 1]
```diff
@@ -19,8 +19,7 @@ run_shellcheck_on () {
     local file
     file="$1"
     echo "Shellcheck: ${file}"
-    # TODO: add -f diff when shellcheck version allows (ubuntu > bionic)
-    shellcheck -x "$file"
+    shellcheck -f diff -x "$file"
```
Comment (phlax) [inline tools/code_format/check_shellcheck_format.sh:22]: this will fail until shellcheck is updated

**Comment by phlax** [comment]
@mattklein123 i think the envoy-build-tools PR (envoyproxy/envoy-build-tools#103) didnt land yet

**Comment by mattklein123** [comment]
Hopefully this will work with the new image that is about to get built.

/wait

**Pull request merged**